beetsplug/absubmit.py

```diff
@@ -0,0 +1,159 @@
+"""Calculate acoustic information and submit to AcousticBrainz.
+"""
+
+from __future__ import division, absolute_import, print_function
+
+import hashlib
+import json
+import os
+import subprocess
+import tempfile
+
+import distutils
+import requests
+
+from beets import plugins
+from beets import util
+from beets import ui
+
+
+class ABSubmitError(Exception):
+    """Base exception for all excpetions this plugin can raise."""
+
+
+class FatalABSubmitError(ABSubmitError):
+    """Raised if the plugin is not able to start."""
+
+
+class AnalysisABSubmitError(ABSubmitError):
+    """Raised if analysis of file fails."""
+
+
+class SubmitABSubmitError(ABSubmitError):
+    """Raised if submitting data fails."""
+
+
+def call(args):
+    """Execute the command and return its output.
+
+    Raise a AnalysisABSubmitError on failure.
+    """
+    try:
+        return util.command_output(args)
+    except subprocess.CalledProcessError as e:
+        raise AnalysisABSubmitError(
+            u"{0} exited with status {1}".format(args[0], e.returncode)
+        )
+
+
+class AcousticBrainzSubmitPlugin(plugins.BeetsPlugin):
+
+    def __init__(self):
+        super(AcousticBrainzSubmitPlugin, self).__init__()
+
+        self.config.add({'extractor': u''})
+
+        self.extractor = self.config['extractor'].as_str()
+        if self.extractor:
+            self.extractor = util.normpath(self.extractor)
+            # Expicit path to extractor
+            if not os.path.isfile(self.extractor):
+                raise FatalABSubmitError(
```

Comment:
This should probably be a `beets.ui.UserError`, right? That leads to a nicely-formatted user-visible error message.